tasks/vagrant.rb

```diff
@@ -47,29 +37,38 @@ def get_vagrant_dir(platform, vagrant_dirs, i = 0)
   platform_dir
 end
 
-def configure_ssh(platform, ssh_config_path)
-  command = "vagrant ssh-config > #{ssh_config_path}"
-  run_local_command(command, @vagrant_env)
-  ssh_config = Net::SSH::Config.load(ssh_config_path, 'default')
-  case platform
-  when %r{debian.*|ubuntu.*}
-    restart_command = 'service ssh restart'
-  when %r{centos.*}
-    restart_command = 'systemctl restart sshd.service'
+def configure_remoting(platform, remoting_config_path)
+  if platform_uses_ssh(platform)
+    command = "vagrant ssh-config > #{remoting_config_path}"
+    run_local_command(command, @vagrant_env)
+    remoting_config = Net::SSH::Config.load(remoting_config_path, 'default')
+    case platform
+    when %r{debian.*|ubuntu.*}
+      restart_command = 'service ssh restart'
+    when %r{centos.*}
+      restart_command = 'systemctl restart sshd.service'
+    else
+      raise ArgumentError, "Unsupported Platform: '#{platform}'"
+    end
+    # Pre-configure sshd on the platform prior to handing back
+    Net::SSH.start(
+      remoting_config['hostname'],
+      remoting_config['user'],
+      port: remoting_config['port'],
+      keys: remoting_config['identityfile'],
+    ) do |session|
+      session.exec!('sudo su -c "cp -r .ssh /root/."')
+      session.exec!('sudo su -c "sed -i \"s/.*PermitUserEnvironment\s.*/PermitUserEnvironment yes/g\" /etc/ssh/sshd_config"')
+      session.exec!("sudo su -c \"#{restart_command}\"")
+    end
   else
-    raise ArgumentError, "Unsupported Platform: '#{platform}'"
-  end
-  Net::SSH.start(
-    ssh_config['hostname'],
-    ssh_config['user'],
-    port: ssh_config['port'],
-    keys: ssh_config['identityfile'],
-  ) do |session|
-    session.exec!('sudo su -c "cp -r .ssh /root/."')
-    session.exec!('sudo su -c "sed -i \"s/.*PermitUserEnvironment\s.*/PermitUserEnvironment yes/g\" /etc/ssh/sshd_config"')
-    session.exec!("sudo su -c \"#{restart_command}\"")
+    command = "vagrant winrm-config > #{remoting_config_path}"
+    run_local_command(command, @vagrant_env)
+    remoting_config = Net::SSH::Config.load(remoting_config_path, 'default')
```

Comment:
Does `Net::SSH::Config.load` work with the winrm configuration?

Reply:
Yup! It writes the WinRM config as an SSH config file (since WinRM doesn't _actually_ have config files).